### PR TITLE
Add "Run on protected" option on Actions

### PR DESCRIPTION
**Pull request by sebjulliand**
### Changes
Resolves https://github.com/codefori/vscode-ibmi/issues/1400

This PR adds a new `Run on protected` attribute to Actions:
Capture d’écran du 2023-11-19 16-04-49

If this attribute is enabled, then the Action will be allowed to be selected and to run on read-only members or from protected filters.

Before this PR, no action was allowed to run on protected or read-only targets.

### Checklist
* [x] have tested my change

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -571,6 +571,10 @@
 									"filter",
 									"browser"
 								]								
+							},
+							"runOnProtected": {
+								"type": "boolean",
+								"description": "If enabled, this Action will be allowed to run on protected/read-only targets."
```
Comment (worksofliam) [inline package.json:577]: I think we have used 'readonly' in labels, instead of 'read-only'

File: src/locale/ids/da.ts [hunk 2]
```diff
@@ -269,5 +269,7 @@ export const da: Locale = {
   'actions.workAction.refresh.filter':'Filter',
   'actions.workAction.refresh.filter.description':'Det aktuelle filter bliver opfrisket',
   'actions.workAction.refresh.browser':'Vindue',
-  'actions.workAction.refresh.browser.description':'Hele vinduet bliver opfrisket'
+  'actions.workAction.refresh.browser.description':'Hele vinduet bliver opfrisket',
+  'actions.workAction.runOnProtected':'Run on protected/read-only',
```
Comment (worksofliam) [inline src/locale/ids/da.ts:273]: Same about 'readonly' here.

**Review by worksofliam — CHANGES_REQUESTED** [review]
Other than some label changes, code looks ok to me. Re-request my review when it's ready! This is a neat change.

**Comment by sebjulliand** [comment]
You're right @worksofliam ; the label we used was actually `read only` elsewhere.
@chrjorgensen may I request your expertise for the Danish translation? 😊

**Review by worksofliam — APPROVED** [review]
Works good for me! This is another nice UX change. You can see in this image that I have a locked member open and don't have any Actions available (as told by the error message).

image

In this screenshot you can see that I have one or more actions enabled in protected mode, and the list appears with them in correctly.

image

@sebjulliand Perhaps we should wait for Christian to add the correct translation and also let him test too before this is merged?

Thanks and great stuff!

**Review by worksofliam — APPROVED** [review]
@chrjorgensen Thank you!!

@sebjulliand Over to you for a merge whenever you like 😄

**Review by chrjorgensen — APPROVED** [review]
@sebjulliand Beautiful feature - works in my tests. 😍 

I made the Danish translation (and fixed some typos). Approved! Merge, Captain! 😃

**Comment by sebjulliand** [comment]
Thank you guys! You've been brilliant, as always!

**Comment by m-tyler** [comment]
Thanks a million for this enhancement.  Makes my life easier when I don't have `*ALLOBJ` on my user profile.
A list of my actions I marked to run in protected mode.  I don't really care if the standard CRT* actions are marked up as we have a CMS tool that should create objects outside of the developer environment. 

image

**Pull request merged**